experimental/packages/opentelemetry-sdk-metrics-base/src/export/AggregationTemporality.ts

```diff
@@ -14,10 +14,29 @@
  * limitations under the License.
  */
 
+import { InstrumentType } from '../InstrumentDescriptor';
+
 /**
  * AggregationTemporality indicates the way additive quantities are expressed.
  */
 export enum AggregationTemporality {
   DELTA,
   CUMULATIVE,
 }
+
+export type AggregationTemporalitySelector = (instrumentType: InstrumentType) => AggregationTemporality;
+
+export const CumulativeTemporalitySelector: AggregationTemporalitySelector = () => AggregationTemporality.CUMULATIVE;
+
+export const DeltaTemporalitySelector: AggregationTemporalitySelector = (instrumentType: InstrumentType) => {
+  switch (instrumentType) {
+    case InstrumentType.COUNTER:
+    case InstrumentType.OBSERVABLE_COUNTER:
+    case InstrumentType.HISTOGRAM:
+    case InstrumentType.OBSERVABLE_GAUGE:
+      return AggregationTemporality.DELTA;
+    case InstrumentType.UP_DOWN_COUNTER:
+    case InstrumentType.OBSERVABLE_UP_DOWN_COUNTER:
+      return AggregationTemporality.CUMULATIVE;
```

Comment:
I think it is misleading for the built-in `DeltaTemporalitySelector` to return cumulative temporality. I assume you are following the behavior defined by spec for `OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE`? In this case, I think this selection logic should be moved to the OTLP package.

IMHO, the two built-in selectors provided by sdk-metrics-base should be constantly returning cumulative temporality or delta temporality.

Reply:
The specification says that the delta selector returns cumulative for the non monotonic counters https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk_exporters/otlp.md

Reply:
Yes, it's for the OTLP exporters. So I'm suggesting that this selector should be moved to OTLP packages.

Reply:
AH I misunderstood. I agree

Reply:
Moved these to `OTLPMetricExporterBase`